library/core/src/fmt/num.rs

```diff
@@ -330,10 +330,17 @@ macro_rules! impl_Exp {
                 if subtracted_precision != 0 {
                     let rem = n % 10;
                     n /= 10;
+                    let last_digit_odd = (n % 10) % 2 != 0;
                     exponent += 1;
-                    // round up last digit
-                    if rem >= 5 {
+                    // round up last digit, round to even on a tie
+                    if rem > 5 || (rem == 5 && last_digit_odd) {
                         n += 1;
+                        // if the digit is rounded to the next power
+                        // instead adjust the exponent
+                        if n % 10 == 0 {
```

Comment:
Hmm, doesn't this only tell you that you overflowed the least significant digit? e.g. maybe n=129 rolled up to 130, but that doesn't need any adjustment. For a new power of 10, like n=99 to 100, maybe something like `n.log10() > (n - 1).log10()` would work?